### ISSUE TITLE
JdbcRepository improvement

### ISSUE THREAD

**Issue by laurikimmel** [jirutka/spring-data-jdbc-repository#4]
It's not easy to extend `JdbcRepository` base functionality in sub classes.  All fields are `private` and there are no getters to access them.

Consider scenario 

``` java
@Repository
public class UserRepository extends JdbcRepository<User, String> {

    ...

    public Iterable<User> findByUserName(String userName) {
        // would be easy to implement in case there is access to
        // jdbcOperations and rowMapper in JdbcRepository
        //
        // return getJdbcOperations().query(..., getRowMapper());
    }
}
```

Make private fields in `JdbcRepository` accessible. Change access modifiers to `protected` or add getters.

**Comment by laurikimmel** [comment]
Just discovered - it's duplicate of issue #2.
Still, fix for #2 is not providing access to `RowMapper` and `RowUnmapper` ...

**Comment by gosuvigi** [comment]
I agree, for example, if you want to return a paged list of results based on some criteria you need to introduce a where clause and don't have access to the jdbcOperations or sqlGenerator fields.
